classes/abstracts/ActionScheduler_WPCLI_Command.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * Abstract for WP-CLI commands.
+ */
+abstract class ActionScheduler_WPCLI_Command extends \WP_CLI_Command {
+
+	const DATE_FORMAT = 'Y-m-d H:i:s O';
+
+	/** @var string[] */
+	protected $args;
+
+	/** @var array<string, string> */
+	protected $assoc_args;
+
+	/**
+	 * Construct.
+	 *
+	 * @param string[]              $args       Positional arguments.
+	 * @param array<string, string> $assoc_args Keyed arguments.
+	 */
+	public function __construct( array $args, array $assoc_args ) {
+		if ( ! defined( 'WP_CLI' ) || ! constant( 'WP_CLI' ) ) {
```

Comment:
```suggestion
		if ( ! defined( 'WP_CLI' ) || ! WP_CLI ) ) {
```

Not sure we need the call to `constant`.

Reply:
Not needed, but more explicit.

Reply:
Asking out of interest, you mean it makes it more explicit that `WP_CLI` is a constant and not a `$string`? I'd have thought the primary use-case for `constant` is cases where the constant name is unknown (held in a variable) or is formed dynamically.

Reply:
That is likely the intent of `constant()`, but I've found that sometimes PHPStan gets mad at `WP_CLI`, and wrapping it in `constant()` fixes it. Always wondered if it's because there's both a constant and class with the same name.